data/src/main/java/io/quarkus/qe/data/query/RepositoryQuery.java

```diff
@@ -0,0 +1,116 @@
+package io.quarkus.qe.data.query;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import io.quarkus.hibernate.orm.panache.PanacheQuery;
+import io.quarkus.panache.common.Page;
+import io.quarkus.panache.common.Parameters;
+import io.quarkus.qe.data.RepositoryEntity;
+import io.quarkus.qe.model.QuarkusExtension;
+import io.quarkus.qe.model.requests.RepositoryQueryRequest;
+
+public class RepositoryQuery {
+
+    public static final String FILTER_BRANCH = "byBranch";
+    public static final String FILTER_RELATIVE_PATH = "byRelativePath";
+    public static final String FILTER_REPO_URL = "byRepoUrl";
+    public static final String FILTER_VERSION = "byVersion";
+
+    public static final String PARAM = "param";
+
+    private static final int DEFAULT_PAGE_SIZE = 25;
+    private static final Predicate<RepositoryEntity> ALWAYS_TRUE = r -> true;
+
+    private Predicate<RepositoryEntity> postLoadFilters = ALWAYS_TRUE;
+
+    private PanacheQuery<RepositoryEntity> query;
+
+    private RepositoryQuery(PanacheQuery<RepositoryEntity> query) {
+        this.query = query.page(Page.ofSize(DEFAULT_PAGE_SIZE));
+    }
+
+    public void filterByRequest(RepositoryQueryRequest request) {
+        filterByRepoUrl(request.getRepoUrl());
+        filterByBranch(request.getBranch());
+        filterByRelativePath(request.getRelativePath());
+        filterByVersion(request.getQuarkusVersion());
+        filterByExtensions(request.getExtensions());
+        filterByLabels(request.getLabels());
+    }
+
+    public RepositoryQuery filterByExtensions(List<QuarkusExtension> extensions) {
```

Comment:
Should return void (return value is never used).

Reply:
I've done this way to implement the builder pattern, but It's true that I didn't use it this way. Will return void instead as you suggested,

Reply:
Even though it's not used this way, I prefer keeping the same pattern as the rest of the filters.